src/main/java/staffconnect/model/availability/Availability.java

```diff
@@ -25,40 +43,79 @@ public class Availability {
     public Availability(String availability) {
         requireNonNull(availability);
         checkArgument(isValidAvailability(availability), MESSAGE_CONSTRAINTS);
-        value = strToDayOfWeek(availability);
+        String [] availabilityFields = availability.split(" ");
+        day = parseToDayOfWeek(availabilityFields[0]);
+        startTime = parseToLocalTime(availabilityFields[1]);
+        endTime = parseToLocalTime(availabilityFields[2]);
+        value = day + " " + startTime + " " + endTime;
     }
 
     /**
      * Returns true if a given string is a valid availability.
      */
     public static boolean isValidAvailability(String test) {
-        return test.matches(VALIDATION_REGEX);
+        if (!test.matches(VALIDATION_REGEX)) {
+            return false;
+        }
+
+        String [] availabilityFields = test.split(" ");
+        DayOfWeek testDay = parseToDayOfWeek(availabilityFields[0]);
+
+        if (!isValidLocalTime(availabilityFields[1]) || !isValidLocalTime(availabilityFields[2])) {
+            return false;
+        }
```

Comment:
I think its better to reorder this to the top, with you guard clauses on top and logic below

Reply:
Are you referring to reordering to be on top of String [] availabilityFields = test.split(" ");

Reply:
Oh okay my bad I did not notice the ```test.split(" ")``` there, then i think it is okay to leave it as it is